test/spmd/tensor/test_pointwise_ops.py

```diff
@@ -10,50 +13,182 @@
 
 from spmd.tensor.ops.pointwise_ops import pointwise_rule
 from spmd import DeviceMesh, DTensor
-from spmd.tensor.placement_types import Shard, Replicate, _Partial, DTensorSpec
+from spmd.tensor.placement_types import (
+    Shard,
+    Replicate,
+    _Partial,
+    DTensorSpec,
+    Placement,
+)
 from torch.distributed.distributed_c10d import ReduceOp
 from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
 
 
+def coerce_dtensor(
+    val: Any,
+    device_mesh: DeviceMesh,
+    placements: Sequence[Placement],
+) -> Any:
+    """
+    Recursively coerce (over Sequence and Dict types) Tensors into DTensors.
+
+    :param device_mesh: the DeviceMesh to use.
+    :param placements: the Placement list to use.
+    :return: the transformed structure.
+    """
+    if isinstance(val, Tensor):
+        return DTensor(
+            val,
+            device_mesh=device_mesh,
+            placements=placements,
+            requires_grad=val.requires_grad,
+        )
+
+    if isinstance(val, Sequence):
+        return [
+            coerce_dtensor(
+                x,
+                device_mesh=device_mesh,
+                placements=placements,
+            )
+            for x in val
+        ]
+
+    if isinstance(val, Dict):
+        return {
+            k: coerce_dtensor(
+                v,
+                device_mesh=device_mesh,
+                placements=placements,
+            )
+            for k, v in val.items()
+        }
+
+    return val
+
+
+def collect_dtensor(val: Any) -> Any:
```

Comment:
nit: I would rename the functions to "convert_to_dtensor" and "convert_from_dtensor" for easy readability.

Reply:
Done; `deepcopy_convert_{to,from}_dtensor()`